elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -186,7 +186,7 @@ public void commit() {
 
     @Override
     public <T> Collection filterCollection(Collection collection, Class<T> entityClass, Set<Predicate> predicates) {
-        if (collection instanceof PersistentBag && !predicates.isEmpty()) {
+        if ((collection instanceof AbstractPersistentCollection) && !predicates.isEmpty()) {
```

Comment:
Because it was too restrictive and was hiding `Predicate` construction errors (i.e. we always constructed bad `Predicate`s for `PersistentSet`s). `PersistentBag != PersistentSet`, for instance. However, any sort of collection we can receive that is an `AbstractPersistentCollection` coming from Hibernate should work.